tests/Encoder.js

```diff
@@ -309,12 +309,18 @@ test('Encode namespaced arguments', t => {
 });
 
 test('Encode optional arguments', t => {
-    t.plan(2)
```

Comment:
same requirement as in the other test file

Reply:
@mradkov @marc0olo I don't think it reasonable to implement this, but should I do it? Will we document this feature in SDK afterwards?

Reply:
Can we get finally on the same wave and understand that this is *standalone* library and it's not about SDK only

Reply:
I agree with Dincho that we should see calldata lib as standalone library that could be used independently. @davidyuk can you please elaborate a bit more about what "this" is about? I am really not deep into this topic right now and need to understand what decision we need to take here. need more background to "this" 😅

Reply:
I can shed some light:
I agreed to add support for `undefined` and `null` instead of `None` variant and unboxed inner type of `Some(T)` - that's for option variant only.

However I see this as aliasing or shortcutting or whatever not as canonical structure that this lib should work with.
While we don't have options for the decoder and I agreed to return undefined/T in case of option variant, with the encoder there is no reason to not support the canonical structure of variants: `{VariantCtor: [args]}`